plasmapy/analysis/swept_langmuir/floating_potential.py

```diff
@@ -0,0 +1,303 @@
+          ``factor * array_size`` is taken, where ``array_size`` is the size of
+          ``voltage`` and ``factor = 0.1`` for ``fit_type = "linear"`` and
+          ``0.2`` for ``"exponential"``.
+        - ``min_points = 0`` The entire passed array is fitted.
+        - ``min_points >= 1`` Exact minimum number of points.
+        - ``0 < min_points < 0`` The minimum number of points is taken as
+          ``min_points * array_size``.
+
+    fit_type: str
+        The type of curve to be fitted to the Langmuir trace.  There are two
+        types of curves ``"linear"`` and ``"exponential"`` (Default).  These
+        specified which `FitFunction` class should be applied to the trace.
+
+        +-------------+----------------------------------------------------------+
+        | linear      | `~plasmapy.analysis.fit_functions.Linear`                |
+        +-------------+----------------------------------------------------------+
+        | exponential | `~plasmapy.analysis.fit_functions.ExponentialPlusOffset` |
+        +-------------+----------------------------------------------------------+
+
+    Returns
+    -------
+    vf: `float` or `numpy.nan`
+        The calculated floating potential (in volts).  Returns `numpy.nan` if the
+        floating potential can not be determined.  How :math:`V_f` is calculated
+        depends on the fit function.  This is described in the `root_solve()`
+        method of the relevant fit function (e.g. the
+        :meth:`~plasmapy.analysis.fit_functions.ExponentialPlusOffset.root_solve`
+        method of
+        `~plasmapy.analysis.fit_functions.ExponentialPlusOffset`).
+
+    vf_err: `float` or `numpy.nan`
+        The error associated with the floating potential calculation (in volts).
+        Returns `numpy.nan` if the floating potential can not be determined.
+        Like :math:`V_f`:, the calculation depends on the applied fit function.
+        The `rood_solve()` method also describes how this is calculated.
+
+    rsq: `float`
+        The coefficient of determination (r-squared) value of the fit.  See the
+        documentation of the `rsq` property on the associated fit function
+        (e.g. the
+        `~plasmapy.analysis.fit_functions.ExponentialPlusOffset.rsq`
+        property of
+        `~plasmapy.analysis.fit_functions.ExponentialPlusOffset`).
+
+    func: sub-class of `~plasmapy.analysis.fit_functions.AbstractFitFunction`
+        The callable function :math:`f(x)` repressing the fit and its results.
+
+    islands: `List[slice]`
+        List of `slice` objects representing the indices of the identified
+        crossing-islands.
+
+    indices: `slice`
+        A `slice` object representing the indices of ``voltage`` and ``current``
+        arrays used for the fit.
+
+    """
+    rtn = FloatingPotentialResults(
+        vf=np.nan,
+        vf_err=np.nan,
+        rsq=None,
+        func=None,
+        islands=None,
+        indices=None
+    )._asdict()
```

Comment:
The default values here are the "failure" values.

I do think the `tuple[2:5]` values are necessary.  It's typical to make plots with the fit overlays on your experimental data.  It's a bit of a sanity check because swept Langmuir traces are not typically "ideal."  It also helps for debugging your fits.  I'm also thinking of writing a function that would optimize `vf_err` and `rsq`, just not sure how to do it yet.  Having this info already in the return will help with that optimization.